app-shared-tests/src/main/java/com/appunite/loudius/UniversalWalkThroughAppTest.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2023 AppUnite S.A.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.appunite.loudius
+
+import androidx.compose.ui.test.assertIsDisplayed
+import androidx.compose.ui.test.onNodeWithText
+import androidx.compose.ui.test.performClick
+import com.appunite.loudius.util.IntegrationTestRule
+import com.appunite.loudius.util.waitUntilLoadingDoesNotExist
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+abstract class UniversalWalkThroughAppTest {
```

Comment:
**info**: this is extracted logic from `AbsWalkThroughAppTest` so it can be used in end-to-end test